extensions/resource-deployment/src/ui/deploymentDialog.ts

```diff
@@ -97,31 +114,37 @@ export class DeploymentDialog extends DialogBase {
 		this._toDispose.push(component.onTextChanged(() => { this.variables[fieldInfo.variableName] = component.value; }));
 		fields.push({ title: fieldInfo.label, component: component });
 
-		if (fieldInfo.type === FieldType.Password && fieldInfo.confirmationRequired) {
-			const confirmPasswordComponent = view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({ ariaLabel: fieldInfo.confirmationLabel, inputType: inputType, required: true }).component();
-			fields.push({ title: fieldInfo.confirmationLabel, component: confirmPasswordComponent });
-
-			this._dialogObject.registerCloseValidator((): boolean => {
-				const passwordMatches = component.value === confirmPasswordComponent.value;
-				if (!passwordMatches) {
-					this._dialogObject.message = { level: azdata.window.MessageLevel.Error, text: localize('passwordNotMatch', "{0} doesn't match the confirmation password", fieldInfo.label) };
-				}
-				return passwordMatches;
+		if (fieldInfo.type === FieldType.SQLPassword) {
+			const invalidPasswordMessage = localize('invalidPassword', "{0} doesn't meet the password complexity requirement. More information: https://docs.microsoft.com/sql/relational-databases/security/password-policy", fieldInfo.label);
+			const passwordMatchMessage = localize('passwordNotMatch', "{0} doesn't match the confirmation password", fieldInfo.label);
+			this.validators.push((): { valid: boolean, message: string } => {
+				return { valid: this.validateSQLPassword(fieldInfo, component), message: invalidPasswordMessage };
 			});
+			if (fieldInfo.confirmationRequired) {
+				const confirmPasswordComponent = view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({ ariaLabel: fieldInfo.confirmationLabel, inputType: inputType, required: true }).component();
+				fields.push({ title: fieldInfo.confirmationLabel, component: confirmPasswordComponent });
 
-			const checkPassword = (): void => {
-				const passwordMatches = component.value === confirmPasswordComponent.value;
-				if (passwordMatches) {
-					this._dialogObject.message = { text: '' };
-				}
-			};
-
-			this._toDispose.push(component.onTextChanged(() => {
-				checkPassword();
-			}));
-			this._toDispose.push(confirmPasswordComponent.onTextChanged(() => {
-				checkPassword();
-			}));
+				this.validators.push((): { valid: boolean, message: string } => {
+					const passwordMatches = component.value === confirmPasswordComponent.value;
+					return { valid: passwordMatches, message: passwordMatchMessage };
+				});
+
+				const updatePasswordMessage = () => {
+					if (this.validateSQLPassword(fieldInfo, component)) {
```

Comment:
is this check supposed to be optional?  I know SQL Server has the option to enable\disable password validation requirement.

Reply:
good to know that, I will add a flag for that.

Reply:
It looks like it's a per-LOGIN setting (based on https://docs.microsoft.com/en-us/sql/relational-databases/security/password-policy?view=sql-server-2017).

Reply:
the scenarios (create bdc, create docker image) that we support all use sa and the password complexity check is required.